dist/pythonlibs/riotctrl_shell/tests/test_gnrc.py

```diff
@@ -0,0 +1,103 @@
+# Copyright (C) 2020 Freie Universität Berlin
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+import logging
+import os
+import riotctrl_shell.gnrc
+import subprocess
+
+from common import init_ctrl
+
+
+logger = logging.getLogger(__name__)
+
+
+RC = None
```

Comment:
This machinery needs some clarification. I won't block it now but I don't get the need to call a specific test script from here. This seems reversed to me.

At the beginning, I understood the modules under `riotctrl_shell/tests` as modules for unit testing the interactions/parsers. If it's the case then, it would be better to mock some calls.

Reply:
IMHO mocks are no good here, because then we might realize to late, that the parsing of a specific command does not work anymore, because the command itself changed, so we need to use the real deal here. I never called scripts here, but a specific test application in `native`. `tests/shell` and `tests/gnrc_udp` seemed good target for the specific shell interaction classes.

I was thinking to provide a dedicated application for this (e.g. to not depend on `tap` interfaces for GNRC; I ping `::1` in the only real network application here anyway), but in the end lazyness won ^^"

Reply:
I can add clarification as a comment, but I'm not sure what clarification is needed in your opinion.

Reply:
(I did not change the tests or which application is used with them, I only changed how and when the `riotctrl` is initialized)

Reply:
> we might realize to late, that the parsing of a specific command does not work anymore, because the command itself changed

That's the purpose of test no ? If the output is changing, then it's normal to expect the test that is testing the output will fail.

Reply:
Yes... then I'm confused why you want a mock? Or were you referring to a dedicated application for these tests?

Reply:
I think there's some confusion on both side :)
In fact, I would use riotctlr and expecially the ICMPv6 interaction/parser from the script of applications that requires it. Code that is under `riotctrl_shell/tests` should only be there for testing the internals of the interactions/parsers modules themselves, so maybe they would need some mock (no real terminal is required in fact). For parser, that could just be injecting test strings and verify that the parser does the job correctly.

Reply:
Ahhh okay, yes I can do that ^^"

Reply:
Ok, I wrote now a very simple mock for `RIOTCtrl` to test the functions and used that in the tests and used legitimate results from the `gnrc_udp` test as constant input strings for testing parsing.

Reply:
Thanks, that looks good!